include/up-cpp/transport/builder/UAttributesBuilder.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _UATTRIBUTESBUILDER_
+#define _UATTRIBUTESBUILDER_
+
+#include <string>
+#include <spdlog/spdlog.h>
+#include <up-cpp/uuid/factory/Uuidv8Factory.h>
+#include <up-core-api/uattributes.pb.h>
+#include <up-core-api/uuid.pb.h>
+
+namespace uprotocol::utransport {
+
+class UAttributesBuilder {
+    private:
+        uprotocol::v1::UAttributes attributes_;
+
+    public:
+        UAttributesBuilder() {
+            attributes_.Clear();
+        }
+
+        UAttributesBuilder(uprotocol::v1::UUri source,
+                           const uprotocol::v1::UUID& id, 
+                           uprotocol::v1::UMessageType type, 
+                           uprotocol::v1::UPriority priority) {
+
+            attributes_.Clear();
+            attributes_.mutable_id()->CopyFrom(id);
+            attributes_.mutable_source()->CopyFrom(source);
+            attributes_.set_type(type);
+            attributes_.set_priority(priority);
+        }
+
+        UAttributesBuilder& setToken(const std::string& token) {
```

Comment:
Methods need documentation, CoPilot can autogenerate them for you, please use it.